### PR TITLE
examples/deploy/update-agent.yaml: make sure update-agent runs as root

### LINKED ISSUE #13: Rebooting node fails because D-Bus rejects the command
```
Jan 13 15:35:36 ip-10-0-40-8 dbus[1293]: [system] Rejected send message, 2 matched rules; type="method_call", sender=":1.33" (uid=65534 pid=31293 comm="/bin/update-agent ") interface="com.coreos.update1.Manager" member="GetStatus" error name="(unset)" requested_reply="0" destination="com.coreos.update1" (uid=0 pid=1292 comm="/usr/sbin/update_engine -foreground -logtostderr ")
```

This might be related to change introduced in 9098c0bdf0745854f800afd76f54defdd602f3c7, where we switched the user we use to run the agent container.

CC @dongsupark

### PR DESCRIPTION
**Pull request by invidian**
Otherwise it may not be allowed to perform a reboot on the host.

Closes #13

Signed-off-by: Mateusz Gozdek <mateusz@kinvolk.io>

### CODE REVIEW

File: examples/deploy/update-agent.yaml [hunk 1]
```diff
@@ -45,6 +45,11 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        # Update agent must run as UID, which is allowed to reboot nodes via logind using D-Bus interface.
```
Comment (pothos) [inline examples/deploy/update-agent.yaml]:
```suggestion
        # Update agent must run with a UID that is allowed to reboot nodes via logind using the D-Bus interface.
```

Comment (invidian) [reply examples/deploy/update-agent.yaml]: Thanks! Fixed.

**Comment by pothos** [comment]
I got this error in the pod logs. Was it seen before?

```
I0114 16:23:21.969410       1 agent.go:251] Indicating a reboot is needed
I0114 16:24:24.218646       1 agent.go:155] Checking if node is already unschedulable
I0114 16:24:24.230590       1 agent.go:171] Setting annotations map[string]string{"flatcar-linux-update.v1.flatcar-linux.net/agent-made-unschedulable":"true", "flatcar-linux-update.v1.flatcar-linux.net/reboot-in-progress":"true"}
I0114 16:24:24.253670       1 agent.go:184] Marking node as unschedulable
I0114 16:24:24.265428       1 agent.go:192] Getting pod list for deletion
I0114 16:24:24.300626       1 agent.go:201] Deleting 0 pods
I0114 16:24:24.300813       1 agent.go:225] Node drained, rebooting
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x11130e3]

goroutine 29 [running]:
github.com/kinvolk/flatcar-linux-update-operator/pkg/updateengine.(*Client).ReceiveStatuses(0xc0002714a0, 0xc0002f91a0, 0xc00007e660)
	/go/src/github.com/kinvolk/flatcar-linux-update-operator/pkg/updateengine/client.go:99 +0x243
created by github.com/kinvolk/flatcar-linux-update-operator/pkg/agent.(*Klocksmith).watchUpdateStatus
	/go/src/github.com/kinvolk/flatcar-linux-update-operator/pkg/agent/agent.go:292 +0xeb

```

**Comment by invidian** [comment]
I think yes:
- https://github.com/coreos/container-linux-update-operator/issues/184
- https://github.com/coreos/container-linux-update-operator/issues/93

**Pull request merged**